<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1102</width>
    <height>1041</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="1" column="0">
     <widget class="QPushButton" name="prev_button">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="text">
       <string>Previous &lt;&lt;</string>
      </property>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QPushButton" name="next_button">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="text">
       <string>Next &gt;&gt;</string>
      </property>
     </widget>
    </item>
    <item row="0" column="0" colspan="2">
     <widget class="QStackedWidget" name="stackedWidget">
      <property name="currentIndex">
       <number>4</number>
      </property>
      <widget class="QWidget" name="page_system_check">
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QLabel" name="title">
          <property name="maximumSize">
           <size>
            <width>16777215</width>
            <height>50</height>
           </size>
          </property>
          <property name="text">
           <string>System Check</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="checkBox">
          <property name="text">
           <string>The machine is empty</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="checkBox_2">
          <property name="text">
           <string>The bins are empty</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton">
          <property name="text">
           <string>confirm</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="page_connect">
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QLabel" name="title_2">
          <property name="text">
           <string>Connect to devices</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="status_robot">
          <property name="text">
           <string>robot unconnnected</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="status_arduino">
          <property name="text">
           <string>arduino unconnected</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="status_camera">
          <property name="text">
           <string>camera unconnected</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="button_connect">
          <property name="text">
           <string>connect everything</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="page_select_subassembly">
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="1" column="1">
         <widget class="QLineEdit" name="search_box"/>
        </item>
        <item row="2" column="0" colspan="2">
         <widget class="QTreeWidget" name="treeWidget">
          <property name="animated">
           <bool>true</bool>
          </property>
          <column>
           <property name="text">
            <string>Subassemblies</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>No. of types</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>No. of items</string>
           </property>
          </column>
          <item>
           <property name="text">
            <string>Engine</string>
           </property>
           <property name="text">
            <string>?</string>
           </property>
           <property name="text">
            <string>297</string>
           </property>
           <property name="flags">
            <set>ItemIsDragEnabled|ItemIsDropEnabled|ItemIsUserCheckable|ItemIsEnabled</set>
           </property>
           <item>
            <property name="text">
             <string>X-1</string>
            </property>
            <property name="text">
             <string>?</string>
            </property>
            <property name="text">
             <string>100</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>X-50</string>
            </property>
            <property name="text">
             <string>?</string>
            </property>
            <property name="text">
             <string>100</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>X-99</string>
            </property>
            <property name="text">
             <string>10</string>
            </property>
            <property name="text">
             <string>97</string>
            </property>
           </item>
          </item>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="selected_title">
          <property name="text">
           <string>Selected engine/subassembly:</string>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QLabel" name="selected_status">
          <property name="text">
           <string>engine/subassembly</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0" colspan="2">
         <widget class="QLabel" name="title_3">
          <property name="text">
           <string>Select subassembly</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="search_title">
          <property name="text">
           <string>Search:</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="page_subassembly_overview">
       <layout class="QGridLayout" name="gridLayout_3">
        <item row="1" column="0">
         <widget class="QLabel" name="selected_title_2">
          <property name="text">
           <string>Selected engine/subassembly</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLabel" name="selected_status_2">
          <property name="text">
           <string>engine/subassembly</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0" colspan="3">
         <widget class="QTableWidget" name="tableWidget">
          <property name="alternatingRowColors">
           <bool>true</bool>
          </property>
          <attribute name="horizontalHeaderCascadingSectionResizes">
           <bool>false</bool>
          </attribute>
          <attribute name="horizontalHeaderStretchLastSection">
           <bool>true</bool>
          </attribute>
          <attribute name="verticalHeaderStretchLastSection">
           <bool>false</bool>
          </attribute>
          <row>
           <property name="text">
            <string>1</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>2</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>3</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>4</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>5</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>6</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>7</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>8</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>9</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>10</string>
           </property>
          </row>
          <column>
           <property name="text">
            <string>bolt type</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>amount in subassembly</string>
           </property>
          </column>
          <item row="0" column="0">
           <property name="text">
            <string>V647P23B</string>
           </property>
          </item>
          <item row="0" column="1">
           <property name="text">
            <string>10</string>
           </property>
          </item>
          <item row="1" column="0">
           <property name="text">
            <string>M59557-10</string>
           </property>
          </item>
          <item row="1" column="1">
           <property name="text">
            <string>10</string>
           </property>
          </item>
          <item row="2" column="0">
           <property name="text">
            <string>M59557-16</string>
           </property>
          </item>
          <item row="2" column="1">
           <property name="text">
            <string>10</string>
           </property>
          </item>
          <item row="3" column="0">
           <property name="text">
            <string>M59557-20</string>
           </property>
          </item>
          <item row="3" column="1">
           <property name="text">
            <string>10</string>
           </property>
          </item>
          <item row="4" column="0">
           <property name="text">
            <string>NAS1802-3-6</string>
           </property>
          </item>
          <item row="4" column="1">
           <property name="text">
            <string>13</string>
           </property>
          </item>
          <item row="5" column="0">
           <property name="text">
            <string>NAS1802-3-7</string>
           </property>
          </item>
          <item row="5" column="1">
           <property name="text">
            <string>10</string>
           </property>
          </item>
          <item row="6" column="0">
           <property name="text">
            <string>NAS1802-3-8</string>
           </property>
          </item>
          <item row="6" column="1">
           <property name="text">
            <string>10</string>
           </property>
          </item>
          <item row="7" column="0">
           <property name="text">
            <string>NAS1802-3-9</string>
           </property>
          </item>
          <item row="7" column="1">
           <property name="text">
            <string>10</string>
           </property>
          </item>
          <item row="8" column="0">
           <property name="text">
            <string>NAS1802-4-07</string>
           </property>
          </item>
          <item row="8" column="1">
           <property name="text">
            <string>4</string>
           </property>
          </item>
          <item row="9" column="0">
           <property name="text">
            <string>NAS6305-10</string>
           </property>
          </item>
          <item row="9" column="1">
           <property name="text">
            <string>10</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="0" column="0" colspan="3">
         <widget class="QLabel" name="title_4">
          <property name="text">
           <string>TextLabel</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="page_machine">
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <layout class="QGridLayout" name="gridLayout_4">
          <item row="3" column="0">
           <widget class="QLabel" name="selected_title_3">
            <property name="text">
             <string>selected engine/subassembly:</string>
            </property>
           </widget>
          </item>
          <item row="5" column="0" colspan="2">
           <widget class="QProgressBar" name="progressBar">
            <property name="value">
             <number>0</number>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QLabel" name="selected_status_3">
            <property name="text">
             <string>engine/subassembly</string>
            </property>
           </widget>
          </item>
          <item row="6" column="0" colspan="2">
           <widget class="QTableWidget" name="tableWidget_2">
            <property name="alternatingRowColors">
             <bool>true</bool>
            </property>
            <attribute name="horizontalHeaderCascadingSectionResizes">
             <bool>false</bool>
            </attribute>
            <attribute name="horizontalHeaderStretchLastSection">
             <bool>true</bool>
            </attribute>
            <attribute name="verticalHeaderStretchLastSection">
             <bool>false</bool>
            </attribute>
            <row>
             <property name="text">
              <string>1</string>
             </property>
            </row>
            <row>
             <property name="text">
              <string>2</string>
             </property>
            </row>
            <row>
             <property name="text">
              <string>3</string>
             </property>
            </row>
            <row>
             <property name="text">
              <string>4</string>
             </property>
            </row>
            <row>
             <property name="text">
              <string>5</string>
             </property>
            </row>
            <row>
             <property name="text">
              <string>6</string>
             </property>
            </row>
            <row>
             <property name="text">
              <string>7</string>
             </property>
            </row>
            <row>
             <property name="text">
              <string>8</string>
             </property>
            </row>
            <row>
             <property name="text">
              <string>9</string>
             </property>
            </row>
            <row>
             <property name="text">
              <string>10</string>
             </property>
            </row>
            <column>
             <property name="text">
              <string>bolt type</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>count</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>expected count</string>
             </property>
            </column>
            <item row="0" column="0">
             <property name="text">
              <string>V647P23B</string>
             </property>
            </item>
            <item row="0" column="2">
             <property name="text">
              <string>10</string>
             </property>
            </item>
            <item row="1" column="0">
             <property name="text">
              <string>M59557-10</string>
             </property>
            </item>
            <item row="1" column="2">
             <property name="text">
              <string>10</string>
             </property>
            </item>
            <item row="2" column="0">
             <property name="text">
              <string>M59557-16</string>
             </property>
            </item>
            <item row="2" column="2">
             <property name="text">
              <string>10</string>
             </property>
            </item>
            <item row="3" column="0">
             <property name="text">
              <string>M59557-20</string>
             </property>
            </item>
            <item row="3" column="2">
             <property name="text">
              <string>10</string>
             </property>
            </item>
            <item row="4" column="0">
             <property name="text">
              <string>NAS1802-3-6</string>
             </property>
            </item>
            <item row="4" column="2">
             <property name="text">
              <string>13</string>
             </property>
            </item>
            <item row="5" column="0">
             <property name="text">
              <string>NAS1802-3-7</string>
             </property>
            </item>
            <item row="5" column="2">
             <property name="text">
              <string>10</string>
             </property>
            </item>
            <item row="6" column="0">
             <property name="text">
              <string>NAS1802-3-8</string>
             </property>
            </item>
            <item row="6" column="2">
             <property name="text">
              <string>10</string>
             </property>
            </item>
            <item row="7" column="0">
             <property name="text">
              <string>NAS1802-3-9</string>
             </property>
            </item>
            <item row="7" column="2">
             <property name="text">
              <string>10</string>
             </property>
            </item>
            <item row="8" column="0">
             <property name="text">
              <string>NAS1802-4-07</string>
             </property>
            </item>
            <item row="8" column="2">
             <property name="text">
              <string>4</string>
             </property>
            </item>
            <item row="9" column="0">
             <property name="text">
              <string>NAS6305-10</string>
             </property>
            </item>
            <item row="9" column="2">
             <property name="text">
              <string>10</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QPushButton" name="start_machine_button">
            <property name="text">
             <string>start sorting machine</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QPushButton" name="stop_machine_button">
            <property name="text">
             <string>stop</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="Line" name="line_2">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <widget class="QLabel" name="label_img_camera">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>480</width>
              <height>384</height>
             </size>
            </property>
            <property name="frameShape">
             <enum>QFrame::Box</enum>
            </property>
            <property name="text">
             <string>Live camera feed</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="Line" name="line">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_img_bolt">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>480</width>
              <height>384</height>
             </size>
            </property>
            <property name="frameShape">
             <enum>QFrame::Box</enum>
            </property>
            <property name="text">
             <string>Detected bolt image</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="detected_bolt_type">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>320</width>
              <height>0</height>
             </size>
            </property>
            <property name="text">
             <string>Detected bolt</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1102</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
